src/diary.cpp

```diff
@@ -0,0 +1,791 @@
+        std::string headtext = string_format( _( "Entry: %d/%d, %s, %s" ),
+                                              opend_page + 1, pages.size(),
+                                              to_string( get_page_ptr()->turn ),
+                                              ( opend_page != 0 ) ? string_format( _( "%s%s%d minutes since last entry" ),
+                                                      ( days > 0 ) ? string_format( _( "%d days, " ), days ) : "",
+                                                      ( hours > 0 ) ? string_format( _( "%d hours, " ), hours ) : "",
+                                                      minutes ) : "" );
+
+        return headtext;
+    }
+    return "";
+}
+
+void diary::death_entry()
+{
+    bool lasttime = query_yn( _( "Open diary for the last time?" ) );
+    if( lasttime ) {
+        show_diary_ui( this );
+    }
+    export_to_txt( true );
+}
+
+diary::diary()
+{
+    owner = get_avatar().name;
+}
+void diary::set_page_text( std::string text )
+{
+    get_page_ptr()->m_text = text;
+}
+
+void diary::new_page()
+{
+    diary_page *page = new diary_page();
+    page -> m_text = " ";
+    page -> turn = calendar::turn;
+    page -> kills = g ->get_kill_tracker().kills;
+    page -> npc_kills = g->get_kill_tracker().npc_kills;
+    avatar *u = &get_avatar();
+    page -> mission_completed = mission::to_uid_vector( u->get_completed_missions() );
+    page -> mission_active = mission::to_uid_vector( u->get_active_missions() );
+    page -> mission_faild = mission::to_uid_vector( u->get_failed_missions() );
+    page -> male = u->male;
+    page->strength = u->get_str_base();
+    page->dexterity = u->get_dex_base();
+    page->intelligence = u->get_int_base();
+    page->perception = u->get_per_base();
+    page->traits = u->get_mutations( false );
+    const auto spells = u->magic->get_spells();
+    for( const spell *spell : spells ) {
+        const spell_id &id = spell->id();
+        const int lvl = spell->get_level();
+        page-> known_spells[id] = lvl;
+    }
+    page-> bionics = u->get_bionics();
+    for( Skill &elem : Skill::skills ) {
+
+        int level = u->get_skill_level_object( elem.ident() ).level();
+        page->skillsL.insert( { elem.ident(), level } );
+    }
+    page -> known_profs = u->_proficiencies->known_profs();
+    page -> learning_profs = u->_proficiencies->learning_profs();
+    page->max_power_level = u->get_max_power_level();
+    diary::pages.push_back( page );
+}
+
+void diary::delete_page()
+{
+    if( opend_page < static_cast<int>( pages.size() ) ) {
+        pages.erase( pages.begin() + opend_page );
+        set_opend_page( opend_page - 1 );
+    }
+}
+
+void diary::export_to_txt( bool lastexport )
+{
+    std::ofstream myfile;
+    std::string path = lastexport ? PATH_INFO::memorialdir() : PATH_INFO::world_base_save_path();
+    path += "\\" + owner + "s_diary.txt";
+    myfile.open( path );
```

Comment:
1. You'd better use `/` as path separator because it's more portable across different operating systems
2. Please use `cata::ofstream myfile( fs::u8path( [std::string path here] ) )` to open files, because avatar names are in UTF-8 encoding and we have this `cata::[i/o]fstream` wrapper to handle UTF-8 path on Windows.